pkg/pillar/zedcloud/authen.go

```diff
@@ -300,14 +302,60 @@ func signAuthData(sigdata []byte, cert tls.Certificate) ([]byte, error) {
 		}
 		log.Debugf("r.bytes %d s.bytes %d\n", len(r.Bytes()),
 			len(s.Bytes()))
-		sigres = r.Bytes()
-		sigres = append(sigres, s.Bytes()...)
+		sigres = RSCombinedBytes(r.Bytes(), s.Bytes())
 		log.Debugf("signAuthData: ecdas sigres (len %d): %x\n",
 			len(sigres), sigres)
 	}
 	return sigres, nil
 }
 
+// RSCombinedBytes - combine r & s into fixed length bytes
+func RSCombinedBytes(rBytes, sBytes []byte) []byte {
+	rsize := len(rBytes)
+	ssize := len(sBytes)
+	size := rsize
+	if ssize > rsize {
+		size = ssize
+	}
+	if size%8 > 0 {
+		size = getUpperMultiOf8(size)
+	}
```

Comment:
Can't we have the callers use ecdsakeyBytes(pub) to determine a length and pass that to this function and avoid this heuristics for computing some length?